tests/sentry/workflow_engine/migration_helpers/test_migrate_rule_action.py

```
@@ -890,3 +922,435 @@ def test_azure_devops_migration_malformed(self):
+                        "default": "20001",
+                        "type": "select",
+                        "updatesForm": True,
+                    },
+                    {
+                        "name": "issuetype",
+                        "label": "Issue Type",
+                        "default": "8",
+                        "type": "select",
+                        "choices": [["1", "Defect"], ["8", "Story"]],
+                        "updatesForm": True,
+                        "required": True,
+                    },
+                ],
+                "description": "This will be generated from the Sentry Issue details.",
+                "project": "20001",
+                "issuetype": "8",
+                "reporter": "user123",
+                "uuid": "22222222-2222-2222-2222-222222222222",
+            },
+        ]
+
+        actions = build_notification_actions_from_rule_data_actions(action_data)
+        self.assert_actions_migrated_correctly(actions, action_data, "integration", None, None)
+
+    def test_jira_server_action_migration_malformed(self):
+        action_data: list[dict[str, Any]] = [
+            # Missing required fields
+            {
+                "uuid": "12345678-90ab-cdef-0123-456789abcdef",
+                "id": "sentry.integrations.jira_server.notify_action.JiraServerCreateTicketAction",
+            },
+            # Empty additional fields
+            {
+                "integration": "123456",
+                "id": "sentry.integrations.jira_server.notify_action.JiraServerCreateTicketAction",
+                "project": "10001",
+                "issuetype": "1",
+                "reporter": "user123",
+                "uuid": "11111111-1111-1111-1111-111111111111",
+                "priority": "",
+                "components": [],
+                "fixVersions": [],
+            },
+        ]
+
+        actions = build_notification_actions_from_rule_data_actions(action_data)
```

not related to this PR, but is it intentional that this helper function fails silently?

---

yes the plan is to only migrate actions that we *can* migrate. i have a log incase we can't so we can go back and debug if its something wrong on our end or just broken config from the start.

---

the plan is to only migrate actions that are configured correctly in case there were actions configured in the old models that were destined to fail. we log those errors to audit after migration to make sure it seems in order.